src/meta.jl

```diff
@@ -1,27 +1,54 @@
-function _tf(ex)
-    @assert ex.head == Symbol("=")
-    name = ex.args[1]
-    call = copy(ex.args[2])
-    @assert call.head == :call
-
-    # Special-case :get_variable, which takes `name` as its first argument instead
-    # of a keyword argument
-    if call.args[1] == :get_variable
-        insert!(call.args, 2, string(name))
-    else
-        if length(call.args) >=2 && isa(call.args[2], Expr) && call.args[2].head == :parameters
-            params = call.args[2]
-        else
-            params = Expr(:parameters)
-            insert!(call.args, 2, params)
-        end
-        push!(params.args, Expr(:kw, :name, string(name)))
-    end
+using MacroTools
+
+# Use Holy traits to make if something is a known Op or not
+abstract OpRegistration
+immutable RegisteredOp<:OpRegistration end
+immutable NotRegisteredOp<:OpRegistration end
+
+is_registered_op(::DataType) = NotRegisteredOp() # By default nothing is registered
+
+
+macro op(f)
```

Comment:
We could.
Or we could bring it back when it is needed.
Or we could dynamically generate it using `methods(is_registered_op)`